app/assets/javascripts/editor.js

```diff
@@ -54,14 +53,19 @@ $E = {
     // preview
     $E.preview = ($D.selected).find('.comment-preview').eq(0);
   },
+  isRichTextEditor: function(url) {
+    // this RegEx matches three different cases where the legacy editor is still used:
+    //   1. /wiki/new
+    //   2. /wiki/{wiki name}/edit
+    //   3. /features/new
+    const legacyEditorPath = RegExp(/\/(wiki|features)(\/[^\/]+\/edit|\/new)/);
+    return !legacyEditorPath.test(url); // if we're not on one of these pages, we are using the rich-text editor.
+  },
   // wraps currently selected text in textarea with strings a and b
-  wrap: function(a,b,args) {
-    // this RegEx is: /wiki/ + any char not "/" + /edit
-    const isWikiCommentPage = (/\/wiki\/[^\/]+\/edit/).test(window.location.pathname);
-    // we don't need to refresh $E's values if we're on a page with a single comment form, ie. /wiki/new or /wiki/edit
-    if (window.location.pathname !== "/wiki/new" && !isWikiCommentPage && $D.selected) {
-      this.refresh();
-    }
+  wrap: function(a, b, args) {
+    // we only refresh $E's values if we are on a page using the rich-text editor (most pages).
+    // the legacy editor pages only have one editor form, unlike pages with multiple comments.
+    if (this.isRichTextEditor(window.location.pathname)) { this.refresh(); }
     var len = $E.textarea.val().length;
     var start = $E.textarea[0].selectionStart;
     var end = $E.textarea[0].selectionEnd;
```

Comment:
@jywarren codeclimate won't let this pass because of 'cognitive complexity' in the wrap function... I think the rules are stricter now since the recent changes in CI.

Honestly, I have a feeling that it's the older stuff later on in the method that's holding this back (conditionals like `args && args['fallback']`, a nested if statement inside `if (args && args['newline'])`. I'm not very familiar with this code yet, can you take a look and let me know if you have any ideas to simplify things here so this can pass?

Reply:
@jywarren Never mind! I got it to work. Tests are passing now too.

Ready to merge!